src/app/story/directives/player.component.ts

```diff
@@ -1,122 +1,157 @@
-  ) {
-    this.tabSub = tab.model.subscribe((s: StoryModel) => {
-      this.setFields(s);
+  shouldUseFeeder: boolean;
+  previewingUnpublished: boolean;
+  loadError: string;
+
+  previewUrl: string;
+  previewSafe: SafeResourceUrl;
+  copyUrl: string;
+  copyIframe: string;
+
+  constructor(tab: TabService, private sanitizer: DomSanitizer) {
+    this.tabSub = tab.model.subscribe((story: StoryModel) => {
+      this.story = story;
+      story.getSeriesDistribution('podcast').subscribe(distDoc => {
+        this.shouldUseFeeder = distDoc ? true : false;
+        this.fromStory(story);
+        if (distDoc) {
+          this.fromFeeder(story, new DistributionModel(story.parent, distDoc));
+        }
+      });
     });
   }
 
-  setFields(story: StoryModel) {
-    this.story = story;
-    this.title = story.title || 'Placeholder title';
-    this.subtitle = story.shortDescription || 'Placeholder subtitle';
-    this.subscriptionUrl = ''; // currently not in place
-
-    if (this.story.images.length > 0) {
-      let img = this.story.images.find(img => !img.isDestroy);
-      if (img) { this.imageUrl = img.enclosureHref; }
-    }
+  ngOnDestroy(): any {
+    this.tabSub.unsubscribe();
+  }
 
-    if (this.story.versions.length > 0) {
-      let firstAudio = this.story.versions[0].files.find(file => !file.isDestroy);
-      if (firstAudio) { this.audioUrl = firstAudio.enclosureHref; }
+  ngDoCheck() {
+    let hasCmsParams = this.title && this.subtitle && this.audioUrl && this.imageUrl && this.subscriptionUrl;
+    let hasFeederParams = this.feedUrl && this.episodeGuid;
+    if (this.shouldUseFeeder === false && hasCmsParams) {
+      this.previewingUnpublished = false;
+      this.buildPlayer(false, false);
     }
-
-    if (this.story.doc && this.story.doc['_links']['prx:series']) {
-      this.subscriptionUrl = this.story.doc['_links']['prx:series'].href;
+    if (this.shouldUseFeeder === true && hasFeederParams) {
+      if (this.story.isPublished()) {
+        this.previewingUnpublished = false;
+        this.buildPlayer(true, true);
+      } else {
+        this.previewingUnpublished = true;
+        this.buildPlayer(false, true); // preview via cms params
+      }
     }
   }
 
-  get paramString() {
-    let str: string[] = [];
+  fromStory(story: StoryModel) {
+    story.loadRelated().subscribe(() => {
+      this.title = story.title || 'Placeholder title';
+      this.subtitle = story.shortDescription || 'Placeholder subtitle';
+      if (story.parent && story.parent['title']) {
+        this.subtitle = story.parent['title'];
+      }
 
-    str.push(`tt=${this.encode(this.title)}`);
-    str.push(`ts=${this.encode(this.subtitle)}`);
-    str.push(`ua=${this.encode(this.audioUrl)}`);
-    str.push(`ui=${this.encode(this.imageUrl)}`);
-    str.push(`us=${this.encode(this.subscriptionUrl)}`);
-    return str.join('&');
-  }
+      // some defaults in case we can't find data
+      this.subscriptionUrl = 'http://www.prx.org';
```

Comment:
Hmm... yeah, that makes sense.  Good sense!

The non-feeder player is also a bit of a TODO... since it doesn't let you select which audio-version or image to use.  And it can't play > 1 mp3 file in an audio-version.  But i figured those edge cases are pretty low priority.